src/components/Payroll/PayrollOverview/PayrollOverviewPresentation.tsx

```diff
@@ -1,74 +1,470 @@
+              title: t('tableHeaders.deductions'),
+              render: (employeeCompensations: EmployeeCompensations) => (
+                <Text>
+                  {formatCurrency(
+                    employeeCompensations.deductions?.reduce(
+                      (acc, deduction) => acc + deduction.amount!,
+                      0,
+                    ) ?? 0,
+                  )}
+                </Text>
+              ),
+            },
+            {
+              title: t('tableHeaders.reimbursements'),
+              render: (employeeCompensations: EmployeeCompensations) => (
+                <Text>{formatCurrency(getReimbursements(employeeCompensations))}</Text>
+              ),
+            },
+            {
+              title: t('tableHeaders.employeeTaxes'),
+              render: (employeeCompensations: EmployeeCompensations) => (
+                <Text>
+                  {formatCurrency(
+                    employeeCompensations.taxes?.reduce(
+                      (acc, tax) => (tax.employer ? acc : acc + tax.amount),
+                      0,
+                    ) ?? 0,
+                  )}
+                </Text>
+              ),
+            },
+            {
+              title: t('tableHeaders.employeeBenefits'),
+              render: (employeeCompensations: EmployeeCompensations) => (
+                <Text>
+                  {formatCurrency(
+                    employeeCompensations.benefits?.reduce(
+                      (acc, benefit) => acc + (benefit.employeeDeduction ?? 0),
+                      0,
+                    ) ?? 0,
+                  )}
+                </Text>
+              ),
+            },
+            {
+              title: t('tableHeaders.payment'),
+              render: (employeeCompensations: EmployeeCompensations) => (
+                <Text>{formatCurrency(employeeCompensations.netPay ?? 0)}</Text>
+              ),
+            },
+          ]}
+          data={payrollData.employeeCompensations!}
+        />
+      ),
+    },
+  ]
 
   return (
     <Flex flexDirection="column" alignItems="stretch">
       <Flex justifyContent="space-between">
-        <Heading as="h1">{t('pageTitle', { startDate: 'Jul 5', endDate: 'Jul 18, 2025' })}</Heading>
+        <div>
+          <Heading as="h1">{t('pageTitle')}</Heading>
+          <Text>{getPayrollOverviewTitle({ payPeriod: payrollData.payPeriod, locale, t })}</Text>
+        </div>
         <Flex justifyContent="flex-end">
-          <Button title={t('buttons.editTitle')} onClick={onEdit} variant="secondary">
-            {t('buttons.edit')}
-          </Button>
-          <Button title={t('buttons.submitTitle')} onClick={onSubmit}>
-            {t('buttons.submit')}
+          <Button onClick={onEdit} variant="secondary">
+            {t('editCta')}
           </Button>
+          <Button onClick={onSubmit}>{t('submitCta')}</Button>
         </Flex>
       </Flex>
+      {/* TODO: when is this actually saved? */}
```

Comment:
Clarity from aaron that within the flow after PayrollConfiguration calculation, we should display the alert for progress saved. If it is being used as a standalone block, or accessed from the payroll history (for payroll summary) we would not show the alert. This is now ticketed here https://gustohq.atlassian.net/browse/GWS-5565